zerver/tests/test_openapi.py

```diff
@@ -537,78 +537,84 @@ def test_openapi_arguments(self) -> None:
                 # accepted by every view function in arguments_map.
                 accepted_arguments = set(arguments_map[function_name])
 
-                regex_pattern = p.pattern.regex.pattern
-                for url_format, url_params in regex_helper.normalize(regex_pattern):
-                    url_pattern = "/" + url_format % {param: f"{{{param}}}" for param in url_params}
-
-                    if "intentionally_undocumented" in tags:
-                        self.ensure_no_documentation_if_intentionally_undocumented(
-                            url_pattern, method
-                        )
-                        continue
+                with self.subTest(function_name):
```

Comment:
The nesting here is a bit crazy; is there a good pace to split out a function for some part of this?

Reply:
Yeah makes sense. Extracted a helper right before subTest.